src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2543,18 +2543,30 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
             $retryOptions = $scopeConfig['retry_failed'] ?? ['enabled' => false];
             unset($scopeConfig['retry_failed']);
 
+
+            $httpClientTransport = new Reference('http_client.transport');
+
+            if ($responseFactoryId = $scopeConfig['mock_response_factory'] ?? null) {
+
+
+                $container->register('http_client.mock_client.' . $name, MockHttpClient::class)
+                    ->setDecoratedService($httpClientTransport, null, -10)  // lower priority than TraceableHttpClient (5)
+                    ->setArguments(
+                        $responseFactoryId === true ? [] : [new Reference($responseFactoryId)]);
```

Comment:
this looks good to me but I don't get the `true` case, what is it for? we don't have it for the general mock factory

Reply:
It is if I want to have a mock client here but I do not want to configure the responsefactory as a service but just do it in the test. I considered having a separate attribute for it, i.e. `mock: true` and then having the mock_response_factory be an optional attribute if needed. 

Maybe being able to just say "mock this" and then add responses in the client is the main use case - i.e. that mock should be a separate parameter to make it more clear.

Reply:
Got it. Let's do the same for the global factory?

Reply:
Good idea. So add a mock: bool to both the global factory and the specific factory?

Reply:
@nicolas-grekas : ping :) I just wanted to get a reaction to the last question before working on the PR. Also: Is it possible to get this into the 6.x series?